### PR TITLE
Move build files and remove submodule

### PR DESCRIPTION
**Pull request by wkal-pubnub**

### CODE REVIEW

File: .github/workflows/release/js-pre-npm-build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+echo "Build Chat SDK module artifacts"
```
Comment (marcin-cebo) [inline .github/workflows/release/js-pre-npm-build.sh:1]: What about adding JS to this text?

File: .github/workflows/release/js-pre-npm-build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+pushd ..
+./gradlew :jsNodeProductionLibraryDistribution
```
Comment (marcin-cebo) [inline .github/workflows/release/js-pre-npm-build.sh:3]: Is jsNodeProductionLibraryDistribution task defined somewhere in our code?

File: build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubJavaLibraryPlugin.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import org.gradle.api.plugins.quality.CheckstylePlugin
+import org.gradle.api.tasks.SourceSetContainer
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.jvm.toolchain.JavaLanguageVersion
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.getByType
+import org.gradle.kotlin.dsl.withType
+
+class PubNubJavaLibraryPlugin : Plugin<Project> {
```
Comment (marcin-cebo) [inline build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubJavaLibraryPlugin.kt:18]: Do we need this file ?

File: build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubTestPlugin.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.withType
+import org.gradle.testing.jacoco.plugins.JacocoPlugin
+import org.gradle.testing.jacoco.tasks.JacocoReport
+
+class PubNubTestPlugin : Plugin<Project> {
+    override fun apply(target: Project) {
+        with(target) {
+            apply<JavaLibraryPlugin>()
```
Comment (marcin-cebo) [inline build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubTestPlugin.kt:16]: Is this plugin required here?

File: build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubTestPlugin.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+            // Tests
+            tasks.named<Test>("test").configure { test ->
+                test.failFast = true
+                test.exclude("**/contract/*.class")
+                test.useJUnitPlatform()
+            }
+
+            tasks.withType<JacocoReport>().configureEach { task ->
+                task.reports {
```
Comment (marcin-cebo) [inline build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubTestPlugin.kt:27]: You could add here
  task.dependsOn(tasks.named("test")) 
to generate report after tests run 
?

File: build-logic/ktlint-custom-rules/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+plugins {
+    `java-library`
```
Comment (marcin-cebo) [inline build-logic/ktlint-custom-rules/build.gradle.kts:2]: Do we need java in KMP?

File: js-chat/rollup.config.mjs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        format: "cjs",
+      },
+      {
+        file: pkg.module,
+        format: "esm",
+      },
+    ],
+    plugins: [
+      terser()
+//       commonjs()
```
Comment (marcin-cebo) [inline js-chat/rollup.config.mjs:21]: This comment is related to content of `fixtures` folder. What is content of `fixtures` folder used for?

**Review by marcin-cebo — APPROVED** [review]

File: .github/workflows/release/js-pre-npm-build.sh (hunk 1 above)
Comment (wkal-pubnub) [reply .github/workflows/release/js-pre-npm-build.sh:1]: done

File: build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubJavaLibraryPlugin.kt (hunk 3 above)
Comment (wkal-pubnub) [reply build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubJavaLibraryPlugin.kt:18]: removed

File: build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubTestPlugin.kt (hunk 5 above)
Comment (wkal-pubnub) [reply build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubTestPlugin.kt:27]: removed this entire file

File: build-logic/ktlint-custom-rules/build.gradle.kts (hunk 6 above)
Comment (wkal-pubnub) [reply build-logic/ktlint-custom-rules/build.gradle.kts:2]: yes, this is needed - we don't distribute this java library, but it's used internally in the build process

File: js-chat/rollup.config.mjs (hunk 7 above)
Comment (wkal-pubnub) [reply js-chat/rollup.config.mjs:21]: these are files that are used by JS tests
the tests are not currently run automatically, but they are functional and can be run manually. I'm planning to include them in the CI process, but want to do it carefully so it doesn't make our build times even longer

File: build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubTestPlugin.kt (hunk 4 above)
Comment (wkal-pubnub) [reply build-logic/gradle-plugins/src/main/kotlin/com/pubnub/gradle/PubNubTestPlugin.kt]: removed the file

File: .github/workflows/release/js-pre-npm-build.sh (hunk 2 above)
Comment (wkal-pubnub) [reply .github/workflows/release/js-pre-npm-build.sh:3]: it's a built in kotlin task

**Pull request merged**